src/main/java/org/schedulesdirect/api/ZipEpgClient.java

```diff
@@ -309,6 +331,49 @@ protected Program fetchProgram(final String progId) throws IOException {
 		}
 		return p;
 	}
+	
+	@Override
+	protected Artwork[] fetchArtwork(String progId) throws IOException {
+		if(closed)
+			throw new IllegalStateException("Instance has already been closed!");
+		
+		String aId = artworkId(progId);
+		
+		List<Artwork> artworks = artCache.get(aId);
+		
+		if(artworks == null) {
+			artworks = new ArrayList<>();
+			artCache.put(aId, artworks);
+			
+			Path path = vfs.getPath(String.format("artwork/%s.txt", aId));
+			if(Files.exists(path)) {
+				try(InputStream ins = Files.newInputStream(path)) {
+					String data = IOUtils.toString(ins, ZIP_CHARSET.toString());
+					if(data != null) {
+						JSONObject artworkInfo;
+						try {
+							artworkInfo = Config.get().getObjectMapper().readValue(data, JSONObject.class);
+							Object temp = artworkInfo.get("data");
+							if(temp instanceof JSONArray) {
+								JSONArray artworkArr = artworkInfo.getJSONArray("data");
+
+								for(int i=0; i<artworkArr.length(); ++i) {
+									JSONObject awObj = artworkArr.getJSONObject(i);
+									artworks.add(new Artwork(awObj, this));
+								}
+							}
+						} catch(JsonParseException e) {
+							throw new JsonEncodingException(String.format("ZipProgram[%s]: %s", progId, e.getMessage()), e, data);
+						}
+					}
+				} catch (JSONException e) {
+					throw new IOException("JSON error!", e);
+				}
+			}
+		}
+		
+		return artworks.toArray(new Artwork[0]);
+	}
```

Comment:
Grabber is already updated to do this (locally).  I create a pull request for those changes soon.